e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js

```diff
@@ -0,0 +1,450 @@
+import {
+  appBar,
+  collectionTable,
+  createAction,
+  getDashboardCard,
+  getDashboardCardMenu,
+  getDashboardCards,
+  modal,
+  popover,
+  queryBuilderHeader,
+  restore,
+  rightSidebar,
+  setActionsEnabledForDB,
+  summarize,
+  visitDashboard,
+  visualize,
+} from "e2e/support/helpers";
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import { SAMPLE_DB_ID } from "e2e/support/cypress_data";
+
+const { ORDERS_ID } = SAMPLE_DATABASE;
+const PG_DB_ID = 2;
+const PERMISSION_ERROR = "Sorry, you don't have permission to see this card.";
+
+describe("scenarios > dashboard > dashboard back navigation", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+    setActionsEnabledForDB(SAMPLE_DB_ID);
+
+    cy.intercept("POST", `/api/dataset`).as("dataset");
+    cy.intercept("GET", "/api/card/*").as("card");
+    cy.intercept("POST", `/api/card/*/query`).as("cardQuery");
+    cy.intercept("PUT", `/api/card/*`).as("updateCard");
+    cy.intercept("GET", `/api/dashboard/*`).as("dashboard");
+    cy.intercept("POST", `/api/dashboard/*/dashcard/*/card/*/query`).as(
+      "dashcardQuery",
+    );
+  });
+
+  it("should display a back to the dashboard button when navigating to a question", () => {
+    const dashboardName = "Orders in a dashboard";
+    const backButtonLabel = `Back to ${dashboardName}`;
+
+    visitDashboard(1);
+    cy.wait("@dashboard");
+    cy.findByTestId("dashcard").findByText("Orders").click();
+    cy.wait("@cardQuery");
+    cy.findByLabelText(backButtonLabel).should("be.visible");
+    cy.icon("notebook").click();
+    summarize({ mode: "notebook" });
+    popover().findByText("Count of rows").click();
+    cy.findByLabelText(backButtonLabel).should("be.visible");
+    visualize();
+    cy.findByLabelText(backButtonLabel).click();
+    cy.findByTestId("dashboard-header")
+      .findByText(dashboardName)
+      .should("be.visible");
+
+    getDashboardCard().realHover();
+    getDashboardCardMenu().click();
+    popover().findByText("Edit question").click();
+    cy.findByLabelText(backButtonLabel).click();
+    cy.findByTestId("dashboard-header")
+      .findByText(dashboardName)
+      .should("be.visible");
+
+    appBar().findByText("Our analytics").click();
+    cy.findByTestId("collection-table").findByText("Orders").click();
+    cy.findByLabelText(backButtonLabel).should("not.exist");
+  });
+
+  it("should display a back to the dashboard button in table x-ray dashboards", () => {
+    const cardTitle = "Sales per state";
+    cy.visit(`/auto/dashboard/table/${ORDERS_ID}`);
+    cy.wait("@dataset");
+
+    getDashboardCards()
+      .filter(`:contains("${cardTitle}")`)
```

Comment:
What is the purpose of those `.filter(":contains...` throughout this PR?
Could the same be accomplished with `findByText` (perhaps with `exact: false`)?

Reply:
No, the goal is to find a card with this title. AFAIK there are no other built-in methods for that.